engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/output/table/base/DefaultTextExtractor.java

```diff
@@ -390,7 +390,17 @@ protected void processParagraphChilds( final ParagraphRenderBox box ) {
         if ( lineBox.getStaticBoxLayoutProperties().isPreserveSpace() == false ) {
           addSoftBreak();
         } else {
-          addEmptyBreak();
+
+          RenderableText renderableText = (RenderableText) lineBox.getFirstChild();
```

Comment:
You cannot assume that the first child will be a renderable text element. It could also be a renderbox, or an image for instance. You have to be more defensive here. In case there are more than one child elements, ask yourself: Do you want to test the first child of the line to determine the type of space or the last child? 

I'd say recursively search the linebox for the deepest last child and base your decision on that. So if last child is box, search recursively within the box, if last child is not text, assume soft break, else use your logic below.

Reply:
@tmorgner I've updated the PR. Now it recursively searches for the deepest not box last child.
If that element is a RenderableText, it decides between soft or empty break.
If not, it just put a soft break.